vermouth/tests/test_dssp.py

```diff
@@ -30,6 +30,272 @@
+        assert found == expected
+
+    def test_sigle_molecules_cycle_one(self, single_mol_system):
+        """
+        One molecule and a one element sequence to repeat over all residues of
+        the molecule.
+        """
+        sequence = 'A'
+        expected = [sequence] * (5 * 3)
+        processor = dssp.AnnotateResidues('test', sequence)
+        processor.run_system(single_mol_system)
+        found = self.sequence_from_system(single_mol_system, 'test')
+        assert found == expected
+
+
+    def test_multi_molecules_cycle_one(self, multi_mol_system_irregular):
+        """
+        Many molecules and a one element sequence to repeat.
+        """
+        sequence = 'A'
+        expected = [sequence] * (15 * 3)
+        processor = dssp.AnnotateResidues('test', sequence)
+        processor.run_system(multi_mol_system_irregular)
+        found = self.sequence_from_system(multi_mol_system_irregular, 'test')
+        assert found == expected
+
+    @staticmethod
+    @pytest.mark.parametrize('sequence', (
+        'ABC',  # Too short
+        'ABCD',  # Too short, match the length of the first molecule
+        'ABCDEFGHIFKLMNOPQRSTU',  # Too long
+        '',  # Empty
+    ))
+    def test_wrong_length(multi_mol_system_irregular, sequence):
+        """
+        Many molecule and a sequence that has the wrong length raises an error.
+        """
+        processor = dssp.AnnotateResidues('test', sequence)
+        with pytest.raises(ValueError):
+            processor.run_system(multi_mol_system_irregular)
+
+    @staticmethod
+    @pytest.mark.parametrize('sequence', (
+        'ABC',  # Too short
+        'ABCD',  # Too short, match the length of the first molecule
+        'ABCDEFGHIFKLMNOPQRSTU',  # Too long
+        '',  # Empty
+        'ABCDEFGHIJKLMNO',  # Length of all the molecules, without filter
+    ))
+    def test_wrong_length_with_filter(multi_mol_system_irregular, sequence):
+        """
+        Many molecules and a sequence that has the wrong length because of a
+        molecule selector.
+        """
+        # We exclude the second molecule. The filter excludes it based on the
+        # number of nodes, which is 15 because it has 5 residues with 3 nodes
+        # each.
+        processor = dssp.AnnotateResidues(
+            'test', sequence,
+            molecule_selector=lambda mol: len(mol.nodes) != (5 * 3),
+        )
+        with pytest.raises(ValueError):
+            processor.run_system(multi_mol_system_irregular)
+
+    @staticmethod
+    def test_empty_system_empty_sequence():
+        """
+        There are no molecules, but the sequence is empty.
+        """
+        system = vermouth.system.System()
+        sequence = ''
+        processor = dssp.AnnotateResidues('test', sequence)
+        try:
+            processor.run_system(system)
+        except ValueError:
+            pytest.fail('Should not have raised a ValueError.')
```

Comment:
Pointless try/except? A test that raises an error also fails.

Reply:
A test that raises an exception appears as errored, not failed.